### PR TITLE
Update testing setup

### PR DESCRIPTION
**Pull request by LeXofLeviafan**
* removing dependency on `nose`
* simplifying mock-command setup
* adding a `[tests]` [optional dependency](https://packaging.python.org/en/latest/guides/installing-using-pip-and-virtual-environments/?highlight=extras#installing-extras) (for clarity)
* setting up testing on Github CI

### CODE REVIEW

File: ft/ft/internal.py [hunk 1]
```diff
@@ -30,7 +30,7 @@ def ftformat(val):
 
 def try_convert_int(inp):
     # Make sure that the input *only* consists of a number (no added spaces..)
-    if not re.match("^[+-]?\d+$", inp):
+    if not re.match(r"^[+-]?\d+$", inp):
```
Comment (LeXofLeviafan) [inline ft/ft/internal.py:33]: `\d` was causing a deprecation warning here (it should be either `\\d` or within a raw literal (`r"…"`)

File: ft/tests/mock_command.py [hunk 2]
```diff
@@ -1,34 +1,37 @@
-from nose.tools import assert_equals
-
 from ft.internal import ftformat
 from ft.functions import function_list
 
 
-def mock_command(cmd):
+def mock(cmd, input=None, **params):
```
Comment (LeXofLeviafan) [inline ft/tests/mock_command.py:5]: These are optional but I'd say it makes sense to pass these parameters here.
Also makes the tests less imperative.

File: ft/tests/test_filter.py [hunk 3]
```diff
@@ -1,27 +1,22 @@
...
+    cmd = mock(Filter, ("const", ["false"], ["3", "5"]))
+    assert cmd.output() == []
 
 
 def test_filter_const_true():
-    test_filter = mock_command(Filter)
-    test_filter.set_input("const", ["true"], ["3", "5"])
-    test_filter.assert_output(["3", "5"])
+    cmd = mock(Filter, ("const", ["true"], ["3", "5"]))
+    assert cmd.output() == ["3", "5"]
```
Comment (LeXofLeviafan) [inline ft/tests/test_filter.py:12]: This produces very similar error messages to `nose.tools.assert_equals()` (as long as the assertion is made immediately within the test function, and `pytest` is run with `-vv`.

Also makes it somewhat clearer what's happening here, I'd say.

File: setup.py [hunk 4]
```diff
@@ -44,6 +44,9 @@
     include_package_data=True,
     install_requires=REQUIRES,
     python_requires=">=3.5",
+    extras_require={
+        'tests': ['pytest'],
+    },
```
Comment (LeXofLeviafan) [inline setup.py:49]: `pip install .[tests]` pulls in the optional tests dependencies (making it convenient for development).

This also means having dev dependencies listed explicitly in `setup.py`

**Review by sharkdp — APPROVED** [review]
Thank you very much

**Pull request merged**